civievent-widget.php

```diff
@@ -367,6 +278,183 @@ public function widget( $args, $instance ) {
 		}
 	}
 
+	/**
+	 * Parse the "custom_display" JSON and return list of valid entries
+	 *
+	 * @param array $instance
+	 *   The widget instance.
+	 * @return array
+	 *   The valid "custom_display" entries.
+	 */
+	protected function validCustomDisplayFields ( $instance ) {
+		$fields = $this->getFields();
+		$customDisplay = json_decode( empty( $instance['custom_display'] ) ? '{}' : $instance['custom_display'], true );
```

Comment:
I think we have to wrap this in html_entity_deccode(). This is because you can't use actual angle-brackets for things like "greater-than-or-equal-to"; normally this operator would be `>=`, but if you put that in the shortcode, it breaks the page. (See https://codex.wordpress.org/Shortcode_API#HTML which says "The suggested workaround for HTML limitations is to use HTML encoding for all user input, and then add HTML decoding in the custom shortcode handler.").  So users need to use entities, e.g., `&gt;=`, and we need to run json-based attributes like `custom_display` through `html_entity_decode()` to support these operators.